src/transactions/types.ts

```diff
@@ -146,14 +146,42 @@ export type InputScriptData = {
   functionArguments: Array<ScriptFunctionArgumentTypes>;
 };
 
+/**
+ * The data needed to generate an View Function payload
+ */
+export type InputViewFunctionData = {
+  function: MoveFunctionId;
+  typeArguments?: Array<TypeTag | string>;
+  functionArguments: Array<EntryFunctionArgumentTypes | SimpleEntryFunctionArgumentTypes>;
+  abi?: ViewFunctionABI;
+};
+export type InputViewFunctionDataWithRemoteABI = InputViewFunctionData & { aptosConfig: AptosConfig };
+
+export type InputViewFunctionDataWithABI = Omit<InputViewFunctionData, "abi"> & {
```

Comment:
im confused, you omit `abi` but then adding it?

Reply:
Yeah, I stared at this a while, because I didn't understand why I did it before either.

What this essentially does is changes it from "optional" to "non-optional", do you have cool fancy ways to do this?

Reply:
can we update it?